### ISSUE TITLE
Controller support in launcher

### ISSUE THREAD

**Issue by mateowoetam** [PrismLauncher/PrismLauncher#791]
### Role

I play Minecraft on the steamdeck and want to be able to start the game with the controller 

### Suggestion

add controller support on the launcher

### Benefit

I will be able to start a game with a controller on the Steamdeck game and desktop modes 

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

When you open the launcher on the Steamdeck the only way to select a game is on the touchscreen, but in desktop mode the screen is off, therefore it would be great if you could navigate the launcher's menu with a controller, or something with xinput.

**Comment by wanesty** [comment]
yes, that'd be cool,
i'm pretty sure Qt have support for controller inputs but it might be hard to implement if/since it wasn't wrote with it in mind;

Both Yuzu and Citra have controller support and use a Qt gui but those have controller settings, so in addition to writing ui compatibility it would most likely mean need gamepad detection and/or mapping..

**Comment by Scrumplex** [comment]
Qt did have gamepad support, but they removed that API in 6.0. Yuzu uses its own controller backend so we can't really reuse that. I am sure Citra does something similar to Yuzu

**Comment by wanesty** [comment]
> I am sure Citra does something similar to Yuzu

i believe Citra and Yuzu have a lot of (UI) code shared since it's from the same "team"

Also both are GPL-2(cirta) and GPL-3(yuzu) so coping their implementation would be doable but kinda heavy since it most likely means adding controller selection, mapping and customization Plus prolly rewrite a lot of the ui to be compatible with it

**Comment by al12gamer** [comment]
I agree this would be useful

**Comment by ttang4299** [comment]
> Yuzu uses its own controller backend so we can't really reuse that. I am sure Citra does something similar to Yuzu

@Scrumplex Perhaps either team would be receptive if you asked politely? I remember melonDS was able to get ahold of the BIOS from DraStic just by asking. Plus, it's not like they'd want to have exclusive rights to controller implementation anyways.

**Comment by wanesty** [comment]
i completely agree with @ttang4299 however as i mentioned, their QT controller input seem to be sharing a lot of code with the emulator input mapping itself, thus separating the qt part from it might be pretty hard


EDIT :
I just though about it, since the goal is to have better steamdeck compatibility, wouldn't it be more interesting to be able to launch an instance from steamOS, and running in gamescope directly.
This is completely different thing than the issue we're discussing but wouldn't the end result be the same or actually better ?

**Comment by Dragon-Mom** [comment]
Controller support in the Prism Launcher menu would be a huge boon. I like to play games on my TV but with Minecraft there's no way to actually launch instances while using a controller making it very difficult.

**Comment by ttang4299** [comment]
Maybe #183 should be merged into this issue

**Comment by al12gamer** [comment]
I would agree it should

**Comment by ttang4299** [comment]
> EDIT : I just though about it, since the goal is to have better steamdeck compatibility, wouldn't it be more interesting to be able to launch an instance from steamOS, and running in gamescope directly. This is completely different thing than the issue we're discussing but wouldn't the end result be the same or actually better ?

@wanesty Just wanted to add my very delayed two cents here. As someone who switches between instances usually within minutes, having several Steam shortcuts for several instances might not be the best idea for me. It's definitely already possible using the create shortcut function and then the add to Steam function, but optimizing the UI for better controller support very likely has broader benefit, as opposed to spamming a Steam library with different individual instances.

**Comment by Dragon-Mom** [comment]
I also have to mirror ttang4299 here, I use too many instances for shortcuts to be practical especially in big picture mode where they are a bit harder to organize or make look nice.

**Comment by ttang4299** [comment]
@Scrumplex Given Yuzu and Citra are no longer active projects, I assume the controller backend is up for grabs with no consequences now? (as long as nothing Nintendo-related is included, that is)

**Comment by Scrumplex** [comment]
@ttang4299 we won't be able to do that really as we don't want to add the whole controller backend of either. We really just want basic support for controllers, not a fully fledged systeem that allows rebinding and multiple binding profiles.

**Comment by Trial97** [comment]
Until somebody picks this up and implements it here are some suggestions:
- on macos https://joystickmapper.com/
- on linux and windows use https://github.com/AntiMicroX/antimicrox

For steam os here you can find how to use it: https://github.com/AntiMicro/antimicro/wiki/Using-antimicro-on-SteamOS (the docs are from the archived fork but it should work similarly on the newer fork mentioned above)

Here is an example of the config for ps5 controller(for antimicrox).
[ps5.amgp.zip](https://github.com/user-attachments/files/15524401/ps5.amgp.zip)

**Comment by al12gamer** [comment]
Just checking in as I have been busy and haven't played Minecraft in a while. This hasn't been added yet, no?

**Comment by Trial97** [comment]
nope still waiting

**Comment by Qwertie-** [comment]
> wouldn't it be more interesting to be able to launch an instance from steamOS, and running in gamescope directly.

This would solve the problem for most people I'd think. You can export a .desktop shortcut to directly launch an instance, but it didn't seem possible to import that in to steam.

**Comment by ttang4299** [comment]
> This would solve the problem for most people I'd think. You can export a .desktop shortcut to directly launch an instance

@Qwertie- Not sure why you'd settle for 2nd best option when accomodating for controller players. As mentioned before, this would be a messy solution for users who have multiple instances.

**Comment by Qwertie-** [comment]
@ttang4299 I imagine creating a steam compatible shortcut is massively easier than adding controller support to the existing UI. And even if the UI did support controllers, the average user who only has one instance would likely want a direct shortcut anyway.

**Comment by ttang4299** [comment]
@Qwertie- Individual instances as Steam shortcuts already exist though. And I think you underestimate the number of users who have multiple instances (one of the largest appeals to Prism Launcher anyways)

**Comment by Qwertie-** [comment]
@ttang4299 You can't add them to steam, it won't allow .desktop files.

**Comment by ttang4299** [comment]
> @ttang4299 You can't add them to steam, it won't allow .desktop files.

@Qwertie- Don't use .desktop files. Google and Reddit found me this solution in the top comment of this post: https://www.reddit.com/r/SteamDeck/comments/yl4uz5/can_i_launch_a_instance_directly_from_steam_prism/

I think any further discussions regarding the creation of individual instances on Linux for Steam are deserving of its own GitHub issue though.

**Comment by felinusfish** [comment]
Has this been abandoned? [QtGamepadLegacy](https://github.com/pumphaus/qtgamepadlegacy) technically is a thing. It hasn't been updated in two years, but it might be worth taking a look at to see if it could work.

**Comment by dragoncoder047** [comment]
support this - 

mojang's launcher is navigable using a controller, so (since I'm on a mac) pressing the system button on the controller allows me to open mojang's launcher, and I don't have to use a mouse to launch the instance. It would be nice to have this feature parity in Prism too

**Comment by ttang4299** [comment]
Going to bump this again myself. Gamepad support in the launcher would not only at this point benefit Steam Deck and living room PC users as mentioned previously, but also users of handheld PCs where the OEM has not bothered to include any trackpads (ASUS ROG Ally and MSI Claw come to mind here). As more people switch to PC, this would widen the accessibility curve to anyone who is readjusting from console.

**Comment by al12gamer** [comment]
I concur

**Comment by BlazingJimbo** [comment]
bump this
sorry for annoying you trial97 😭

**Comment by drakosoldier298** [comment]
another month later im here to also bymp this. as somone whos turned my old steamdeck into a home console this would be a game changer

**Comment by Leon-Swains** [comment]
Bumping this as well, this feature would be extremely useful for SteamOS PCs. XMCL added gamepad/controller support to the launcher last month and it is a life changing feature.

**Comment by al12gamer** [comment]
I'll join the bump train. I have a few friends who moved from Minecraft to Hytale though, so I may do the same soon.

**Comment by ttang4299** [comment]
@Scrumplex Have there been any discussions internally about where this would lie in the roadmap? As an enhancement that’s been sitting in the pipeline for over 3 years now, it’s not a feature I’m asking to rush, but it would be nice to get more developer insight, positive or negative.

**Comment by Dragon-Mom** [comment]
Been 2 years since my last post so just here to say again this is still a strongly desired feature lol.

Using Prism in SteamOS would be a lot easier without having to do virtual mouse stuff.

**Comment by Leon-Swains** [comment]
I just wanna say, I REALLY like the idea of #183

**Comment by ttang4299** [comment]
> I just wanna say, I REALLY like the idea of [#183](https://github.com/PrismLauncher/PrismLauncher/issues/183)

I agree, I actually previously suggested merging #183 and #791 together.
